extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java

```diff
@@ -152,5 +152,5 @@ public class DataSourceReactiveRuntimeConfig {
      * are initiated.
      */
     @ConfigItem
-    public Map<String, String> additionalProperties;
+    public Optional<Map<String, String>> additionalProperties;
```

Comment:
Thanks for trying but this is not the right fix :).

The problem is that for named datasources, the objects here need to be initialized as we might in some cases instantiate a new instance of this class without passing through the config framework.

The problem is more global than that. I see quite a lot of issues with this, unfortunately. I will fix it globally.

Reply:
No worries. I figured the map being `null` instead of empty was probably an invalid state.